nuvla/job_engine/job/actions/utils/deployment_utils.py

```diff
@@ -18,27 +21,35 @@
 from ....connector.k8s_driver import get_kubernetes_local_endpoint
 
 
-def get_connector_name(deployment):
+HELM_CONNECTOR_KIND = 'helm'
+HELM_APP_SUBTYPE = 'application_helm'
+
+
+def get_connector_name(deployment: Union[dict, CimiResource]):
     if Deployment.is_component(deployment):
         return 'docker_service'
 
-    elif Deployment.is_application(deployment):
+    if Deployment.is_application(deployment):
         is_compose = Deployment.is_compatibility_docker_compose(deployment)
         return 'docker_compose' if is_compose else 'docker_stack'
 
-    elif Deployment.is_application_kubernetes(deployment):
+    if Deployment.is_application_kubernetes(deployment):
         return 'kubernetes'
 
+    if Deployment.subtype(deployment) == HELM_APP_SUBTYPE:
+        return HELM_CONNECTOR_KIND
+
     subtype = Deployment.subtype(deployment)
     raise ValueError(f'Unsupported deployment subtype "{subtype}"')
```

Comment:
Because, all `elif`-s return from the function and if none was visited, then the implied else is the `raise`, I simply turned it into n independent branches. But yes - all are mutually exclusive and from the readability point of view keeping `elif`-s is probably better. Reverting.